products/alp/main.pm

```diff
@@ -16,6 +17,7 @@ init_main();
 my $distri = testapi::get_required_var('CASEDIR') . '/lib/susedistribution.pm';
 require $distri;
 testapi::set_distribution(susedistribution->new());
+testapi::set_distribution(DistributionProvider->provide());
```

Comment:
we just need to set it once, the one 'provided' is derived class of susedistribution.

Reply:
updated